File: SerialPort.cpp
#include "SerialPort.h"

SerialPort::SerialPort(char* devicename,int baudrate)
{
    deviceName=devicename;
    baudRate=baudrate;
}

SerialPort::~SerialPort()
{

}
/**
 * This function sets serial port options.
*/
void SerialPort::initialize(){
    // Clear all the size bits
    tty.c_cflag&=~CSIZE;
    // Clear stop field only one stop bit is used
    tty.c_cflag&=~CSTOPB;
    // Number of bits per byte is 8
    tty.c_cflag&=~CS8;
    // Disable RTS/CTS hardware flow control 
    tty.c_cflag&=~CRTSCTS;
    // Disable canonical mode
    tty.c_lflag &= ~ICANON;
    // Disable Echo
    tty.c_lflag &= ~ECHO;
    // Turn off s/w flow ctrl
    tty.c_iflag &= ~(IXON | IXOFF | IXANY);
    // Set baud rate
    cfsetispeed(&tty, baudRate);
    cfsetospeed(&tty, baudRate);
    // Open port
    serial_port = open(deviceName, O_RDWR);
}
/**
 * This function take message as a paramater and writes to port
*/
void SerialPort::writeData(char msg[]){
    write(serial_port, msg, sizeof(msg));
}

File: serialPort.py
import serial
from datetime import datetime
import time

class SerialPort():
    def __init__(self,baudrate=9600,port="/dev/ttyUSB0"):
        self.serial_port=serial.Serial()
        self.serial_port.baudrate=baudrate
        self.serial_port.port=port

    def isport_working(self,message):
        try:
            self.serial_port.open()
            self.serial_port.write(bytes(message,"utf-8"))
            self.serial_port.close()
            return True
        except:
            raise "Port is not working"

    def send_datetime(self):
        self.serial_port.open()
        while(True):
            now=bytes(datetime.now().strftime("%H:%M:%S")+"\n","utf-8")
            self.serial_port.write(now)
            time.sleep(1)

sp=SerialPort()
print(sp.isport_working("OK"))
sp.send_datetime()
